### PR TITLE
9 uxfunctional improvement ideas(1,3,4,5,6)

### LINKED ISSUE #9: UX/Functional Improvement Ideas
Taking some notes so that we don't forget later:

- [x] 1. Improve UX for download/save workflow, by potentially providing a hyperlink to gov.uk's page so that the users are able to grab CSV urls more easily.
- [ ] 2. Expose download/save workflow when the local storage is populated too. Right now only way to update the csv is to reinstall the plugin. This is suboptimal :D
- [x] 3. Pluralize `N companies` matching your filter on the search tab.
- [x] 4. Format the number of matching companies (I guess with commas; e.g `84249` -> `84,249`).
- [x] 5. Render company names using smaller text.
- [x] 6. Render `DownloadCSVPanel` with padding
- [ ] 7. Add a last-modified-date field into `CompanyData` type. This field could be used to make decisions about whether to update the local storage.

### PR DESCRIPTION
**Pull request by irmk323**
From the https://github.com/irmk323/sponsor-checker/issues/9 , completed task 1,3,4,5,6. 
2 and 7 would be done in the separated branch.

### CODE REVIEW

File: src/popup/style.scss [hunk 1]
```diff
@@ -4,6 +4,9 @@
     display: flex;
     flex-direction: row-reverse;
 }
+.mantine-Table-td > p {
+    font-size: 14px;
```
Comment (oal55) [inline src/popup/style.scss]: Here I think we could future-proof ourselves a little. The current selector will technically apply to _all_ mantine tables we render in the plugin. Right now there's only one, so the above selector works nicely.

2 ideas:
- Using the size prop in table data (in Popup.tsx L126) to render smaller text
- If we need finer control, it looks like there are [a few mantine-endorsed ways](https://mantine.dev/styles/styles-api/) to go about customizing (I am not super opinionated as to which one we should use). We could use either the `styles`, or `classes` props for the <Table ... /> element.

File: src/popup/Popup.tsx [hunk 2]
```diff
@@ -175,15 +177,27 @@ const DownloadCSVPanel: React.FC<{setcompanyNames: (companyNames: string[]) => v
...
+            </Text>
             <Input
                 placeholder="Download URL for the csv"
                 value={url}
                 onChange={e => setUrl(e.target.value)}
                 mt="md"
             />
             <div className="reverse-flex">
-                <Button loading={isLoading} size="xs" variant="filled" onClick={handleDownloadClicked}>Save CSV</Button>
+                <Button m={10} loading={isLoading} size="xs" variant="filled" onClick={handleDownloadClicked}>Save CSV</Button>
```
Comment (oal55) [inline src/popup/Popup.tsx:197]: What does the `m` prop do 👀?

File: src/popup/Popup.tsx [hunk 3]
```diff
@@ -120,6 +120,7 @@ const CompaniesPanel: React.FC<{ companyNames: string[] }> = ({ companyNames })
         }
         return companyNames.filter(name => name.toLowerCase().includes(lowerCaseText));
     }, [companyNames, searchText]);
+    const formattedCompanyCount = new Intl.NumberFormat().format(filteredCompanyNames.length);
```
Comment (oal55) [inline src/popup/Popup.tsx:123]: This is v neat! Didn't know about Intl ✌️

File: src/popup/Popup.tsx [hunk 4]
```diff
@@ -175,15 +177,27 @@ const DownloadCSVPanel: React.FC<{setcompanyNames: (companyNames: string[]) => v
...
                 placeholder="Download URL for the csv"
                 value={url}
                 onChange={e => setUrl(e.target.value)}
                 mt="md"
             />
             <div className="reverse-flex">
-                <Button loading={isLoading} size="xs" variant="filled" onClick={handleDownloadClicked}>Save CSV</Button>
+                <Button m={10} loading={isLoading} size="xs" variant="filled" onClick={handleDownloadClicked}>Save CSV</Button>
             </div>
         </>
```
Comment (oal55) [inline src/popup/Popup.tsx]: Now that there is a container, we can remove this react fragment.

React enforces that every compondnt returns a single "node", and so fragments are commonly used as wrapper nodes in components where we want to return multiple elements.

They have no effect on the generated HTML. They only exist in the virtual DOM.

**Review by oal55 — COMMENTED** [review]
🚀

File: src/popup/Popup.tsx (hunk 2 above)
Comment (irmk323) [reply src/popup/Popup.tsx:197]: It sets margin around save button :)

File: src/popup/Popup.tsx (hunk 4 above)
Comment (irmk323) [reply src/popup/Popup.tsx]: I see, I removed that fragment.

File: src/popup/style.scss (hunk 1 above)
Comment (irmk323) [reply src/popup/style.scss]: Ah you're right. Changed L126 to  <Text **size="sm"** ... > , then removed this cheeky css.

Initially I was trying to do it by <Table data={tableData} styles={{ tbody: { fontSize: '12px' } }}>  and found it didn't work. Thank you.

File: src/popup/Popup.tsx (hunk 2 above)
Comment (oal55) [reply src/popup/Popup.tsx:197]: ah nice, noted 👌

**Comment by oal55** [comment]
Thank you!

**Pull request merged**